File: Random/2.Isomorphic Strings.js
/*
	Author : Mohamed Yousef 
	Date   : 2022-12-02
*/
let solve=(t,s) => {
    n1=s.length;
    n2=t.length;
    if (n1 !=n2) return false;
	let defaultdict1=new Proxy({},{
		get :(target,Name) => Name in target? target[Name] : 0
	})
	let defaultdict2=new Proxy({},{
		get :(target,Name) => Name in target? target[Name] : 0
	})
	// map s to t
	for(let i=0; i<n1; i++){
		defaultdict1[s[i]]=t[i];
	}
	// map t to s
	for(let i=0; i<n1; i++){
		defaultdict2[t[i]]=s[i];
	}
	// console.log(defaultdict1);
	// console.log(defaultdict2);
	compare1=""
	compare2=""
	for(let i=0; i<n1; i++){
		e=s[i]
		compare1+=defaultdict1[e]
	}
	for(let i=0; i<n1; i++){
		e=t[i]
		compare2+=defaultdict2[e]
	}
	if(compare1==t && compare2==s) return true
	return false
}

File: Random/2.Isomorphic Strings.py
def solve(s,t):
    from collections import defaultdict
    our_dict1 = defaultdict()
    our_dict2=defaultdict()
    n1=len(s)
    n2=len(t)
    if n1!=n2:
        return False
    for i in range(n1):
        our_dict1[t[i]]=s[i]
        our_dict2[s[i]]=t[i]
    compare1=""
    compare2=""
    # print(our_dict1)
    for i in range(n1):
        compare1+=our_dict1[t[i]]
        compare2+=our_dict2[s[i]]
    # print(compare1)
    if compare1==s and compare2==t:
        return True
    return False
print(solve("ffi","fii"))